zerver/views/streams.py

```diff
@@ -246,11 +249,16 @@ def compose_views(request, user_profile, method_kwarg_pairs):
 def remove_subscriptions_backend(
         request: HttpRequest, user_profile: UserProfile,
         streams_raw: Iterable[str]=REQ("subscriptions", validator=check_list(check_string)),
-        principals: Optional[Iterable[str]]=REQ(validator=check_list(check_string), default=None),
+        principals: Optional[Iterable[Union[str, int]]]=REQ(validator=check_variable_type([
+            check_list(check_string), check_list(check_int)]), default=None),
 ) -> HttpResponse:
 
-    removing_someone_else = principals and \
-        set(principals) != set((user_profile.email,))
+    removing_someone_else = False
+    if principals:
```

Comment:
@timabbott We could avoid this conditional by just doing `set(principal_to_user_profile(user_profile, principal) for principal in principals)` more toward the top of the method, but we'd then lose the optimization of `people_to_unsub = set([user_profile])` for the case where `principals` was the empty set.  I think I'm fine with the current code.

Reply:
Yeah, I'm basically fine with this.